openedx/core/djangoapps/discussions/models.py

```diff
@@ -418,3 +417,55 @@ def available_providers(self) -> list[str]:
     @classmethod
     def get_available_providers(cls, context_key: CourseKey) -> list[str]:
         return ProviderFilter.current(course_key=context_key).available_providers
+
+
+class ProgramDiscussionsConfiguration(TimeStampedModel):
+    """
+    Associates a program with a discussion provider and configuration
+    """
+
+    program_uuid = models.CharField(
+        primary_key=True,
+        db_index=True,
+        unique=True,
+        max_length=255,
+        verbose_name=_("Program UUID"),
+    )
+    enabled = models.BooleanField(
+        default=True,
+        help_text=_("If disabled, the discussions in the associated program will be disabled.")
+    )
+    lti_configuration = models.ForeignKey(
+        LtiConfiguration,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        help_text=_("The LTI configuration data for this program/provider."),
+    )
+    provider_type = models.CharField(
+        blank=False,
+        max_length=100,
```

Comment:
again -- can you recheck if you really need 100 characters for storing provider type?

Reply:
Not really. Is it beneficial to be very specific about char length?

Reply:
yes - i believe the db allocate that much of bytes for each value so for known data we need to be specific which is in this case.

Reply:
I have kept it to 50 just in case there comes in a provider that has a long id.